SCClassLibrary/Common/Quarks/QuarksGui.sc

```diff
@@ -29,64 +31,80 @@ QuarksGui {
 		bounds = Window.flipY(Window.availableBounds);
 		window = Window("Quarks", Rect(0, 0, min(bounds.width * 0.75, 1000), bounds.height * 0.9).center_(bounds.center));
 
-		lblCaption = StaticText().font_(GUI.font.new(size:16, usePointSize:true)).string_("Quarks");
+		colors = (
+			error: (bg: Color.fromHexString("#fcdede"), text: Color.fromHexString("#991111")),
+			info: (bg: Color.fromHexString("#e2eef9"), text: Color.fromHexString("#224466")),
+			warn: (bg: Color.fromHexString("#f7ea57"), text: Color.fromHexString("#494620")),
+			success: (bg: Color.fromHexString("#f5ffef"), text: Color.fromHexString("#6cc644")),
+			primary: (bg: Color.fromHexString("#76c656"), text: Color.white),
+			installed: (bg: Color.fromHexString("#76c656"), text: Color.white),
+			default: (bg: Color.white, text: Color.black)
+		);
 
 		btnUpdateDirectory = Button()
-			.states_([["Refresh Quarks directory"]])
-			.toolTip_("Download directory listing from" + Quarks.directoryUrl + "(auto-refreshes every 4 hours)")
+			.states_([["Check for updates"], ["Checking..."]])
+			.toolTip_("Download directory listing from" + Quarks.directoryUrl)
 			.action_({
 				treeView.enabled = false;
-				this.setMsg("Fetching" + Quarks.directoryUrl, \yellow);
-				AppClock.sched(0.2, {
-					protect {
-						model.fetchDirectory(true);
-					} {
-						treeView.enabled = true;
-						this.setMsg("Quarks directory has been updated.", \green);
-						this.update;
-					}
+				this.setMsg("Fetching" + Quarks.directoryUrl, \info);
+				this.checkForUpdates({
+					this.setMsg("Directory has been updated.", \success);
+					treeView.enabled = true;
+					btnUpdateDirectory.value = 0;
+					this.update;
+				}, {
+					this.setMsg();
+					treeView.enabled = true;
+					btnUpdateDirectory.value = 0;
+					this.update;
 				});
 				nil
 			});
 
-		btnQuarksHelp = Button().states_([["Quarks Help"]])
+		btnQuarksHelp = Button().states_([["Help"]])
 			.toolTip_("Open Quarks documentation")
 			.action_({ HelpBrowser.openBrowsePage("Quarks") });
 
-		btnOpenFolder = Button().states_([["Open Quarks Folder"]])
+		btnInstallFolder = Button().states_([["Install a folder"]])
+			.toolTip_("Install classes from a folder on your computer")
+			.action_({ this.installFolder });
+
+		/*btnOpenFolder = Button().states_([["Open Quarks Folder"]])
 			.toolTip_("Open" + model.folder)
-			.action_({ model.openFolder });
+			.action_({ model.openFolder });*/
 
 		btnClear = Button().states_([["Uninstall all"]])
 			.toolTip_("Uninstall all")
 			.action_({ model.clear; this.update; });
 
-		btnLoad = Button().states_([["Load Quarks file"]])
-			.toolTip_("Install a set of quarks from a file")
+		btnSave = Button().states_([["Save Quarks set"]])
```

Comment:
Quarks file -> Quarks set. Because we already have something called a .quarkfile